src/index.tsx

```diff
@@ -0,0 +1,557 @@
+				this.editingArea = instance
+			},
+		};
+
+		if (React.Children.count(children)) {
+			return React.cloneElement(
+				React.Children.only(children)!,
+				properties
+			);
+		}
+
+		return preserveWhitespace ?
+			<pre {...properties}/> :
+			<div {...properties}/>;
+	}
+
+	render() {
+		return (
+			<div
+				id={this.props.id}
+				style={this.props.style}
+				key={this.state.generation}
+				className={`quill ${this.props.className ?? ''}`}
+				onKeyPress={this.props.onKeyPress}
+				onKeyDown={this.props.onKeyDown}
+				onKeyUp={this.props.onKeyUp}
+			>
+				{this.renderEditingArea()}
+			</div>
+		);
+	}
+
+	onEditorChange = (
+		eventName: 'text-change' | 'selection-change',
+		rangeOrDelta: Range | DeltaStatic,
+		oldRangeOrDelta: Range | DeltaStatic,
+		source: Sources,
+	) => {
+		if (eventName === 'text-change') {
+			this.onEditorChangeText?.(
+				this.editor!.root.innerHTML,
+				rangeOrDelta as DeltaStatic,
+				source,
+				this.unprivilegedEditor!
+			);
+		}
+		if (eventName === 'text-change' || eventName === 'selection-change') {
+			this.onEditorChangeSelection?.(
+				rangeOrDelta as RangeStatic,
+				source,
+				this.unprivilegedEditor!
+			);
+		}
+	};
+
+	onEditorChangeText(
+		value: string,
+		delta: DeltaStatic,
+		source: Sources,
+		editor: UnprivilegedEditor,
+	): void {
+		if (!this.editor) return;
+		const currentContents = this.getEditorContents();
+
+		// We keep storing the same type of value as what the user gives us,
+		// so that value comparisons will be more stable and predictable.
+		const nextContents = this.isDelta(currentContents)
+			? editor.getContents()
+			: editor.getHTML();
+
+		if (!this.isEqualValue(nextContents, currentContents)) {
+			// Taint this `delta` object, so we can recognize whether the user
+			// is trying to send it back as `value`, preventing a likely loop.
+			this.lastDeltaChangeSet = delta;
+			this.setState({ value: nextContents });
+			this.props.onChange?.(value, delta, source, editor);
+		}
+	}
+
+	onEditorChangeSelection(
+		nextSelection: RangeStatic,
+		source: Sources,
+		editor: UnprivilegedEditor,
+	): void {
+		if (!this.editor) return;
+		const currentSelection = this.getEditorSelection();
+		const hasGainedFocus = !currentSelection && nextSelection;
+		const hasLostFocus = currentSelection && !nextSelection;
+
+		if (isEqual(nextSelection, currentSelection)) return;
+
+		this.setState({ selection: nextSelection });
+		this.props.onChangeSelection?.(nextSelection, source, editor);
+
+		if (hasGainedFocus) {
+			this.props.onFocus?.(nextSelection, source, editor);
+		} else if (hasLostFocus) {
+			this.props.onBlur?.(currentSelection, source, editor);
+		}
+	}
+
+	focus(): void {
+		if (!this.editor) return;
+		this.editor.focus();
+	}
+
+	blur(): void {
+		if (!this.editor) return;
+		this.setEditorSelection(this.editor, null);
+	}
+}
+
+// Re-export to avoid having to `require(...).default`
+module.exports = ReactQuill;
```

Comment:
It would probably be OK to drop the code supporting CJS as part of the breaking changes, given ESM is on its way into Node, and the compatibility with .default and top-level defaults was broken long ago by Babel